### ISSUE TITLE
[BUG] Error using popup on iOS

### ISSUE THREAD

**Issue by acaliaro** [CommunityToolkit/Maui#1844]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

I show a popup. When I close the popup, I have an exception

```bash
 An error occurred: 'Object reference not set to an instance of an object.'. Callstack: '   at CommunityToolkit.Maui.Core.Views.MauiPopup.SetShadowView(UIView& target)
   at CommunityToolkit.Maui.Core.Views.MauiPopup.ViewDidLayoutSubviews()
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 58
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 94
   at daze.dazeapp.Program.Main(String[] args) in C:\Daze\daze.app.maui\daze.dazeapp\daze.dazeapp\Platforms\iOS\Program.cs:line 12
   at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)'
```

I think the problem is that "target" sometimes is null

	static void SetShadowView(in UIView target)
	{
		if (target.Class.Name is "_UICutoutShadowView")
		{
			target.RemoveFromSuperview();
		}

		if (target.Class.Name is "_UIPopoverDimmingView")
		{
			target.BackgroundColor = UIColor.Black.ColorWithAlpha(0.4f);
		}

		foreach (var view in target.Subviews)
		{
			SetShadowView(view);
		}
	}

maybe a check on "target" and "target.Subviews" should solve the problem...

### Expected Behavior

It should not crash

### Steps To Reproduce

I have a popup that I Use as loader... when I close the popup after some "works", sometimes the exception is thrown

### Link to public reproduction project repository

I haven't one...

### Environment

```markdown
- .NET MAUI CommunityToolkit:8.0.1
- OS:iOS 17.3
- .NET MAUI: 8.0.21
```

**Comment by acaliaro** [comment]
@brminnick sorry but I am not able to create a repo. The problem appears in a very large app. 


The information I sent you on the crash seems quite clear and the null check of the object is also punctual. Do you think the change can be made?
Thanks!

**Comment by brminnick** [comment]
I can't fix the bug if I cannot identify the root cause. I can't identify the root cause without a reproduction.

**Comment by bijington** [comment]
@acaliaro how about trying to make that change locally and see if that solves your problem? Then if we can gain confidence it solves the issue your changes could make it into a PR

**Comment by SumitMandal** [comment]
@brminnick I am also facing this issue on the iPhone 11 or lower. The iOS version does not matter.

Steps to reproduce(1st approach):
1. Open CommunityToolkit.Maui.Sample.sln in the samples folder.
2. Choose the target device as iPhone 11 or lower
3. Run the project
4. Open Flyout window
5. Choose "Views"
6. Choose "Multi Popups Page" card
7. Click "Simple Popup" button
8. Once the Popup opens rotate the device, when you rotate the device for the third time the app will crash.

Steps to reproduce(2nd approach):
1. Open CommunityToolkit.Maui.Sample.sln in the samples folder.
2. Choose the target device as iPhone 11 or lower
3. Run the project
4. Rotate the device one time
5. Open Flyout window
6. Choose "Views"
7. Choose "Multi Popups Page" card
8. Click "Simple Popup" button
9. The app will crash

The reason behind the crash is the PresentationController.ContainerView that is being passed in the SetShadowView method is null.

[https://github.com/CommunityToolkit/Maui/blob/567eee2d1a1b2a3d3d6eba014b877027eaf31623/src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.macios.cs#L55](Code Link)

https://github.com/CommunityToolkit/Maui/assets/5452331/d6bbff50-656a-489f-a53f-f54357c0d9fa

**Comment by MAUIoxo** [comment]
As discussed in [#1864](https://github.com/CommunityToolkit/Maui/issues/1864), I used a simple `Popup` defined like this and created it in Code-behind as can be seen below. It was displayed via `Application.Current?.MainPage?.ShowPopup(myPopup);`:

```xaml
<toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
               xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
               xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
               x:Class="MyApp.SpecialPopup"
               CanBeDismissedByTappingOutsideOfPopup="False">
    <VerticalStackLayout Padding="20"
                         HorizontalOptions="Center"
                         VerticalOptions="Center">
        <Label Text="Welcome to the Fullscreen Popup!"
               HorizontalOptions="Center" />
        <Button Text="Close"
                Clicked="OnCloseButtonClicked"/>
    </VerticalStackLayout>
</toolkit:Popup>
```

And in Code-behind I created a new instance of it and displayed it via:

```cs
var myPopup = new SpecialPopup();
…
Application.Current?.MainPage?.ShowPopup(myPopup);
```

As I changed to my own implementation of a `Popup` using a `ContentPage`, I can't provide a minimal reproducible example. But, you can try with the above setup and then switch from *Portrait* to *Landscape*. When I did this with newer MAUI NIghtly-Builds, I could reproduce it just by switching the orientation

This resulted in `Exceptions` in `SetShadowView(...)` with a `NullReferenceException`

![326288575-f2776053-64ce-4dcc-bb03-870f1436c6f7](https://github.com/CommunityToolkit/Maui/assets/140683727/4095a899-66c4-4f94-b714-a21e51e6d4bf)

![326288654-30c7d215-a904-4bbc-8758-f3c15a1a9345](https://github.com/CommunityToolkit/Maui/assets/140683727/033ab502-5e7a-41af-8089-5401cb96c3f6)

**Comment by cat0363** [comment]
I changed the simulator on iOS 17 and verified whether the problem occurred.

iPhone 15 : OK
iPhone 15 Plus : NG
iPhone 15 Pro : OK
iPhone 15 Pro Max : NG

This issue occurred in Plus and Pro Max simulators.
This issue occurred with the simulator on iPhone 12 and later versions of Plus and Pro Max.
This issue also occurred on iPhone 11 or lower versions.

In addition, in versions earlier than Community Toolkit 8.0.0, the crash does not occur when the device is rotated, but the Popup becomes full screen without the background color becoming transparent.

https://github.com/CommunityToolkit/Maui/assets/125236133/cc51786d-69d2-49bf-bc41-5cb3a1309a25

By applying PR #1615, the above issue will be resolved on iPhone 12 and later simulators other than Plus and ProMax.
However, other simulators will run into issue with PR #1844. Even if you fix it so that the Null Reference Exception does not occur, the problem will not be resolved. 

Even if you specify Start in the Vertical Options and Horizontal Options of Popup, if you rotate the device orientation, it will be displayed full screen and centered. 

https://github.com/CommunityToolkit/Maui/assets/125236133/31f4b99a-573f-4ff2-a02a-6fdb6302414d

Even if you explicitly specify the Popup position in SourceRect, it will not be reflected if the device orientation is Landscape.

[src\CommunityToolkit.Maui.Core\Views\Popup\PopupExtensions.macios.cs]

    mauiPopup.PopoverPresentationController.SourceRect = new CGRect(originX, originY, 0, 0);

It is unclear why the position of Popup is not reflected. It is also unclear why the Popup size covers the entire screen when the device orientation is Landscape...

We need to resolve a Popup size and position issue that occurs when I change the device orientation to Landscape.

**Comment by MAUIoxo** [comment]
Thanks for finding that out! I recognize parts of it also in previous Simulator versions or .NET MAUI versions! I also used a Pro Max here. The FullScreen issue in previous versions when rotating is also very familiar to me!

**Comment by cat0363** [comment]
@MAUIoxo , Thank you for providing the information.
Does anyone know if the same problem occurs on a real device?

**Comment by MAUIoxo** [comment]
I have an iPhone 14 Pro Max and I had a crash of my App with this Bug when changing orientation! Also the problem regarding full screen in previous versions is well known to me on a real device

**Comment by cat0363** [comment]
@MAUIoxo , Thank you for reporting the problem on a real device.
I have found the model to reproduce the problem, but unfortunately I do not have a solution at this time.

**Comment by MAUIoxo** [comment]
Well, not bad for me, I created my own Popup with a simple `ContentPage` which is working fine in my environment

**Comment by cat0363** [comment]
I found the solution for this issue. I will create a PR after tomorrow.
Just adding a few lines of code will solve problem.

**Comment by MAUIoxo** [comment]
Wow, that was fast - do you want to tell us about the problem you found?

**Comment by cat0363** [comment]
@MAUIoxo , I just created a PR, so please take a look at it.

**Comment by erossini** [comment]
In my `.NET8` `MAUI` application, I use the `MAUI CommunityToolkit` version 9.0.1. I have an issue with the Popup that crashes my app. I saw a [bug on GitHub](https://github.com/CommunityToolkit/Maui/issues/1844) that is exactly my issue but there is no answer for it.

I defined my Popup view

    <?xml version="1.0" encoding="utf-8" ?>
	<toolkit:Popup
			x:Class="LanguageInUse.Views.Popups.LoadingPopup"
			xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
			xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
			xmlns:lang="clr-namespace:LanguageInUse"
			xmlns:popups="clr-namespace:LanguageInUse.Views.Popups"
			xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">

		<toolkit:Popup.Resources>
			<Style TargetType="{x:Type popups:LoadingPopup}">
				<Setter Property="Size" Value="300,300" />
				<Setter Property="Color" Value="White" />
				<Setter Property="HorizontalOptions" Value="Center" />
				<Setter Property="VerticalOptions" Value="Center" />
				<Setter Property="CanBeDismissedByTappingOutsideOfPopup" Value="False" />
			</Style>
		</toolkit:Popup.Resources>

		<StackLayout
				Margin="30"
				HorizontalOptions="CenterAndExpand"
				VerticalOptions="CenterAndExpand"
				WidthRequest="350">
			<VerticalStackLayout HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand">
				<ActivityIndicator
						HorizontalOptions="Center"
						IsRunning="True"
						Color="{StaticResource Gray900}" />
				<Label
						x:Name="labelLoading"
						MaximumWidthRequest="240"
						Style="{StaticResource labelLoading}"
						Text="{lang:Translate Loading}"
						WidthRequest="240" />
			</VerticalStackLayout>
		</StackLayout>
	</toolkit:Popup>

When my app is in `Release` mode and it displays the popup on iOS, I get this error

> CommunityToolkit.Maui.Core.Views.MauiPopup.SetShadowView(UIView&
> target)
> CommunityToolkit.Maui.Core.Views.MauiPopup.ViewDidLayoutSubviews()
> ObjCRuntime.Runtime.ThrowException(IntPtr )
> UIKit.UIApplication.UIApplicationMain(Int32 , String[] , IntPtr ,
> IntPtr ) UIKit.UIApplication.Main(String[] , Type , Type )
> LanguageInUse.Program.Main(String[] args)

To display the Popup, I use this line

    var loadingPopup = new LoadingPopup();
    loadingPopup.SetMessage(AppResources.DictionaryWordsLoading);
    loadingPopup.Opened += async (s, e) =>
    {
        try
        {
            // my code
        }
        catch (Exception ex)
        {
        }
        finally
        {
            if (loadingPopup != null)
                await loadingPopup.CloseAsync();
        }
    };
    await Application.Current.MainPage.ShowPopupAsync(loadingPopup);
    loadingPopup = null;

How can I fix it?

**Comment by bijington** [comment]
@erossini this issue is closed, can you please open a new issue?

**Comment by acaliaro** [comment]
@erossini this should be solved with #1892 . @bijington do you have a date to have this fix on a new nuget?

**Comment by bijington** [comment]
> @erossini this should be solved with #1892 . @bijington do you have a date to have this fix on a new nuget?

It was released 4 days ago which is the version @erossini is using
